src/publish-build/index.ts

```diff
@@ -0,0 +1,90 @@
+import * as core from '@actions/core';
+import {
+  S3Client,
+  PutObjectCommand,
+  PutObjectCommandOutput,
+  PutObjectCommandInput,
+} from '@aws-sdk/client-s3';
+import { promises as fs, createReadStream } from 'fs';
+import mime from 'mime';
+import * as path from 'path';
+
+const REQUIRED_ENV_VARS = [
+  'AWS_ACCESS_KEY_ID',
+  'AWS_SECRET_ACCESS_KEY',
+  'AWS_BUCKET',
+  'PROJECT',
+  'SOURCE_DIR',
+  'DESTINATION_DIR',
+  'COMMIT_HASH',
+];
+
+function validateEnvVars(): void {
+  for (const requiredVar of REQUIRED_ENV_VARS) {
+    if (!process.env[requiredVar]) {
+      const errMsg = `Required env variable missing: ${requiredVar}`;
+      core.error(errMsg);
+      throw new Error(errMsg);
+    }
+  }
+}
+
+async function getFileNames(dir: string): Promise<string[]> {
+  const dirents = await fs.readdir(dir, { withFileTypes: true });
+  const files = await Promise.all(
+    dirents.map(async dirent => {
+      const res = path.resolve(dir, dirent.name);
+      return dirent.isDirectory() ? getFileNames(res) : res;
+    }),
+  );
+  return Array.prototype.concat(...files);
+}
+
+async function upload(
+  sourcePath: string,
+  directoryPath: string,
+  fileNames: string[],
+): Promise<PutObjectCommandOutput[]> {
+  const accessKeyId = process.env['AWS_ACCESS_KEY_ID'] ?? '';
```

Comment:
These credentials shouldn't be needed as long as the configure-aws-credentials GitHub Actions is run as a step in the workflow. [Here is an example of this action being used](https://github.com/mongodb/docs-worker-pool/blob/8215c5478ae3a0f9af7a283d172d1c34956c1c2c/.github/workflows/deploy-prd-enhanced-webhooks.yml#L17). This action will export the credentials so that they are available between steps.

 If you do want to use these credentials directly, we should obfuscate them so that they don't get displayed in the case of some error being thrown, or from accidental logging. [Here is an example of obscuring secrets](https://github.com/mongodb/docs-worker-actions/blob/c197b035bcadabcea20206a6622a7193c2b2e868/src/rebuild-parse-cache/src/get-repos.ts#L102).

Reply:
done!